### PR TITLE
[AC-9247]: Change program family query to use rolegrants

### PR DESCRIPTION
**Pull request by kenware**
[AC-9247](https://masschallenge.atlassian.net/browse/AC-9247)

 How to test:
 - Ensure program filter uses program family names from program role grant

### CODE REVIEW

**Review by iankigen — COMMENTED** [review]
Code looks good. 
Waiting for a reindex to finish QA

File: accelerator_abstract/models/base_core_profile.py [hunk 1]
```diff
@@ -486,7 +487,13 @@ def interest_category_names(self):
         return [interest.name for interest in self.interest_categories.all()]
 
     def program_family_names(self):
-        return [pf.name for pf in self.program_families.all()]
+        return self.user.programrolegrant_set.filter(
+            program_role__program__program_status__in=[
+                ACTIVE_PROGRAM_STATUS,
+                UPCOMING_PROGRAM_STATUS
+            ]
```
Comment (iankigen) [inline accelerator_abstract/models/base_core_profile.py:494]: I had missed the comment on the ticket: https://masschallenge.atlassian.net/browse/AC-9247?focusedCommentId=45432

the filter should only care about program role grants for Finalists, Confirmed Mentors, and Alumni. 
Also, program_status filter should not be included (I see no requirement for this in the ticket)

**Review by jonkiparsky — CHANGES_REQUESTED** [review]
Changes look good, test fails when run against development, which is good. 

Two minor changes requested, but this should be good to go once those are addressed.

File: accelerator_abstract/models/base_core_profile.py [hunk 2]
```diff
@@ -486,7 +486,14 @@ def interest_category_names(self):
         return [interest.name for interest in self.interest_categories.all()]
 
     def program_family_names(self):
-        return [pf.name for pf in self.program_families.all()]
+        return self.user.programrolegrant_set.filter(
```
[the remark this answers is not in the document]
Comment (kenware) [reply accelerator_abstract/models/base_core_profile.py:489]: This line is more than 78 characters `program_roles = self.user.programrolegrant_set.filter(program_role__user_role__name__in=[` . so I am using this format

```
program_roles = self.user.programrolegrant_set.filter(
         program_role__user_role__name__in=[BaseUserRole.FINALIST,
                                                                        BaseUserRole.ALUM, 
                                                                        BaseUserRole.MENTOR])
        return program_roles.values_list('program_role__program__program_family__name',
                                                               flat=True)
```

File: accelerator_abstract/models/base_core_profile.py [hunk 3]
```diff
@@ -486,7 +486,12 @@ def interest_category_names(self):
         return [interest.name for interest in self.interest_categories.all()]
 
     def program_family_names(self):
-        return [pf.name for pf in self.program_families.all()]
+        program_roles = self.user.programrolegrant_set.filter(
+            program_role__user_role__name__in=[BaseUserRole.FINALIST,
+                                               BaseUserRole.ALUM,
+                                               BaseUserRole.MENTOR])
+        return list(program_roles.values_list(
+            'program_role__program__program_family__name', flat=True))
```
Comment (iankigen) [inline accelerator_abstract/models/base_core_profile.py:494]: Noticed some redundant program_family names. This should be distinct()

**Review by iankigen — COMMENTED** [review]
From QA
- Active accounts with Finalist role grants appear in the Community directory when the Finalist filter is applied ✅ 

- Active accounts with a Program Role Grant appear in the Community directory when their program family filter is applied ✅ 

- Active accounts with an Alum role grant appear in the Community directory when the Alum filter is applied ✅

**Comment by jonkiparsky** [comment]
@kenware lint issue
```
0.51s$ make code-check
accelerator_abstract/models/base_core_profile.py:494:80: E501 line too long (81 > 79 characters)
Makefile:126: recipe for target 'code-check' failed
make: *** [code-check] Error 123
```

**Review by iankigen — APPROVED** [review]

**Pull request merged**